pkg/agent/apiserver/handlers/ovsflows/handler_test.go

```diff
@@ -169,48 +182,154 @@ func TestServiceFlows(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	testNetworkPolicy := &cpv1beta.NetworkPolicy{}
+func TestNetworkPolicyFlowsSuccess(t *testing.T) {
 	testcases := []testCase{
 		{
-			test:           "Existing NetworkPolicy",
+			testName:       "Existing NetworkPolicy",
 			name:           "np1",
-			namespace:      "ns1",
-			query:          "?networkpolicy=np1&&namespace=ns1",
+			namespace:      "default",
+			policyType:     cpv1beta.K8sNetworkPolicy,
+			query:          "?networkpolicy=np1&namespace=default&type=K8sNP",
 			expectedStatus: http.StatusOK,
 		},
 		{
-			test:           "Non-existing NetworkPolicy",
-			name:           "np2",
-			namespace:      "ns2",
-			query:          "?networkpolicy=np2&&namespace=ns2",
-			expectedStatus: http.StatusNotFound,
+			testName:       "Existing ACNP",
+			name:           "acnp1",
+			policyType:     cpv1beta.AntreaClusterNetworkPolicy,
+			query:          "?networkpolicy=acnp1&type=ACNP",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			testName:       "Existing ANNP",
+			name:           "annp1",
+			namespace:      "default",
+			policyType:     cpv1beta.AntreaNetworkPolicy,
+			query:          "?networkpolicy=annp1&namespace=default&type=ANNP",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			testName:       "Existing ANNP - no type provided",
+			name:           "annp1",
+			namespace:      "default",
+			query:          "?networkpolicy=annp1&namespace=default",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			testName:       "Existing ANP",
+			name:           "anp1",
+			policyType:     cpv1beta.AdminNetworkPolicy,
+			query:          "?networkpolicy=anp1&type=ANP",
+			expectedStatus: http.StatusOK,
 		},
 	}
 	for i := range testcases {
 		tc := testcases[i]
-		npq := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
-		q := aqtest.NewMockAgentQuerier(ctrl)
-		q.EXPECT().GetNetworkPolicyInfoQuerier().Return(npq).Times(1)
-
-		if tc.expectedStatus != http.StatusNotFound {
+		t.Run(tc.testName, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			npq := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+			q := aqtest.NewMockAgentQuerier(ctrl)
 			ofc := oftest.NewMockClient(ctrl)
 			ovsctl := ovsctltest.NewMockOVSCtlClient(ctrl)
-			npq.EXPECT().GetNetworkPolicies(&querier.NetworkPolicyQueryFilter{SourceName: tc.name, Namespace: tc.namespace}).Return([]cpv1beta.NetworkPolicy{*testNetworkPolicy}).Times(1)
-			ofc.EXPECT().GetNetworkPolicyFlowKeys(tc.name, tc.namespace).Return(testFlowKeys).Times(1)
+			npFilter := &querier.NetworkPolicyQueryFilter{
+				SourceName: tc.name,
+				Namespace:  tc.namespace,
+				SourceType: tc.policyType,
+			}
+			q.EXPECT().GetNetworkPolicyInfoQuerier().Return(npq).Times(1)
+			policyTypeToReturn := tc.policyType
+			if policyTypeToReturn == "" {
+				// Test the case where policy type is omitted from the query but the result is not ambiguous
+				policyTypeToReturn = cpv1beta.AntreaNetworkPolicy
+			}
```

Comment:
may be better to have `policyType` *and* `policyTypeToReturn` as fields in the test case definition, to avoid the special case here, but I don't feel very strongly about it

Reply:
@Dyanngg just checking that you didn't address it intentionally?

Reply:
Yes for now, since this is not functionality related. I've updated it locally. Once the CI passes for Jenkins, I'll update the PR again with the changes in UT

Reply:
So to clarify I should wait before I approve & merge the PR?

Reply:
Yes I'll mention you in this thread once that's done, thanks

Reply:
@antoninbas Done. The jenkins tests were passing before I pushed a refactor in UT.